src/server/req_runjob.c

```diff
@@ -1612,17 +1611,12 @@ post_sendmom(struct work_task *pwt)
 			else
 				free_nodes(jobp);
 
-			/*
-			 * If there is a checkpoint, we leave exec_vnode
-			 *    to force that it will run there again;
-			 * Or if stagein was already successful, we leave
-			 *    exec_vnode set as the files are there and
-			 *    we dont want to copy them again;
-			 * ELSE clear exec_vnode, exec_host, etc.
-			 */
+			/* turn off the staged in flag */
+			if (jobp->ji_qs.ji_svrflags & JOB_SVFLG_StagedIn)
+				remove_stagein(jobp);
```

Comment:
If we are unable to remove the stage-in files from the mom, the staged-in flag would still be set.

Reply:
No. it will be removed in remove_stagein method.